### PR TITLE
Updating rule id and logic

### PR DESCRIPTION
**Pull request by veronicagg**
Fixing #1917 and #1921.

### CODE REVIEW

**Comment by veronicagg** [comment]
I've opened a new pr for #2006, to fix history.

File: src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/Validation/anonymous-response-type.json [hunk 1]
```diff
@@ -25,9 +25,15 @@
         "responses": {
           "default": {
             "description": "Unexpected error",
-            "schema": {
+			"schema": {
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/Validation/anonymous-response-type.json]: nit: formatting

File: src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs [hunk 2]
```diff
@@ -29,8 +36,13 @@ public class AnonymousParameterTypes : TypedRule<SwaggerParameter>
...
 
         /// <summary>
-        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// An entity fails this rule if it has a schema, and that schema is an anonymous type
         /// </summary>
-        public override Category Severity => Category.Info;
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public override bool IsValid(SwaggerParameter entity) =>
+            entity?.Schema == null || AnonymousTypesRule.IsValid(entity.Schema);
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs]: No need to check ```entity``` for null, just pass ```entity.Schema``` to ```AnonymousTypesRule``` and check for null there

File: src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs [hunk 3]
```diff
@@ -26,9 +34,12 @@ public class AvoidAnonymousTypes : TypedRule<SwaggerObject>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
-        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// An <paramref name="entity"/> fails this rule if it doesn't have a reference (meaning it's defined inline)
         /// </summary>
-        public override Category Severity => Category.Info;
+        /// <param name="entity">The entity to validate</param>
+        /// <returns></returns>
+        public override bool IsValid(SwaggerObject entity) => (entity?.GetType() == typeof(Schema) &&
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs]: Or, just do ```((Schema)entity.Properties?.Count??0) ==0```

File: src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs [hunk 4]
```diff
@@ -3,22 +3,29 @@
...
 using AutoRest.Core.Logging;
 using AutoRest.Core.Properties;
-using AutoRest.Swagger.Validation.Core;
 using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
 
 namespace AutoRest.Swagger.Validation
 {
-    public class AnonymousParameterTypes : TypedRule<SwaggerParameter>
+    public class AnonymousBodyParameter : TypedRule<SwaggerParameter>
```
Comment (vishrutshah) [inline src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs]: You may move `Id`, `ValidationCategory`, `Severity` & `MessageTemplate ` into top level `AnonymousTypeRule<T>` abstract class that would help in removing 13 for `new AvoidAnonymousTypes()`. upto you not required

File: src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs (hunk 2 above)
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs]: @dsgouda if Schema is null then isValid method should return true, that's why I'm checking for it here.

File: src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs (hunk 3 above)
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs]: @dsgouda Similarly in this case, I'd like IsValid to return true if Properties is null or if there are 0 properties, so I believe I need to keep the check here.

File: src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs (hunk 4 above)
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs]: I'll probably leave as is for now, thanks.

File: src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs (hunk 3 above)
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs]: If properties is null, the expression I suggested will evaluate to 0 and pass the if condition

**Pull request merged**